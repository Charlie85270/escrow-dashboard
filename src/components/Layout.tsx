import * as React from 'react';
import { Box } from '@mui/material';
import Header from './Header';

interface ILayout {
  children: React.ReactNode;
}

const Layout: React.FC<ILayout> = ({ children }): React.ReactElement => (
  <Box
    sx={{
      marginTop: '88px',
    }}
  >
    <Header />
    {children}
  </Box>
);

export default Layout;
