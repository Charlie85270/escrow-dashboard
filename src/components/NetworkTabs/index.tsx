import { Tab, Tabs } from '@mui/material';
import React from 'react';

// import EthereumIcon from '../icons/EthreumIcon';
import HumanIcon from '../Icons/HumanIcon';
// import NeonIcon from '../icons/NeonIcon';
import PolygonIcon from '../Icons/PolygonIcon';

const tabs = [
  { id: 'all', title: 'All Networks', icon: <HumanIcon /> },
  { id: 'polygon', title: 'Polygon', icon: <PolygonIcon /> },
  { id: 'eth-rinkeby', title: 'Ethereum Rinkeby', icon: <PolygonIcon /> },
  { id: 'neonlabs', title: 'NeonLabs', icon: <PolygonIcon /> },
];

export default function NetworkTabs() {
  return (
    <Tabs sx={{ my: '44px' }} value="all">
      {tabs.map(({ id, title, icon }) => (
        <Tab
          key={id}
          value={id}
          icon={icon}
          iconPosition="start"
          label={title}
          sx={{ textTransform: 'none' }}
        />
      ))}
    </Tabs>
  );
}
