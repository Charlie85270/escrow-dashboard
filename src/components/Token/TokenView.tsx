import * as React from 'react';
import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';
import { CardTextBlock, CardLinkBox } from '../Cards';

interface IToken {
  address: string;
  scanner: string;
  transferEventCount?: number | string;
  approvalEventCount?: number | string;
  holders?: number | string;
}

function getTotal(
  transferEventCount: number | string,
  approvalEventCount: number | string
): number | string {
  if (
    typeof transferEventCount === 'string' &&
    typeof approvalEventCount === 'string'
  ) {
    return 'N/A';
  }

  return Number(transferEventCount) + Number(approvalEventCount);
}
export const TokenView: React.FC<IToken> = ({
  address,
  scanner,
  transferEventCount = 'N/A',
  approvalEventCount = 'N/A',
  holders = 'N/A',
}): React.ReactElement => {
  const totalEvents = getTotal(transferEventCount, approvalEventCount);
  const hmtUrl = `${scanner}/address/${address}`;
  return (
    <Card variant="outlined">
      <CardContent>
        <CardLinkBox url={hmtUrl} text={address} header="Token Address" />
        <CardTextBlock title="Token Transfers" value={transferEventCount} />
        <CardTextBlock title="Token Approvals" value={approvalEventCount} />
        <CardTextBlock title="Token Holders" value={holders} />
        <CardTextBlock
          title="Total Number Of Token Events"
          value={totalEvents}
        />
      </CardContent>
    </Card>
  );
};
