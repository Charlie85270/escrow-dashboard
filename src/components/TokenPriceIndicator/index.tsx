import { Box, Typography, useTheme } from '@mui/material';
import React from 'react';

export default function TokenPriceIndicator() {
  const theme = useTheme();
  return (
    <Box
      sx={{
        borderRadius: '4px',
        overflow: 'hidden',
        background: '#f6f7fe',
        display: 'flex',
      }}
    >
      <Box
        sx={{
          background: theme.palette.primary.main,
          fontSize: 12,
          fontWeight: 600,
          padding: '6px 12px',
        }}
      >
        HMT
      </Box>
      <Box sx={{ display: 'flex', alignItems: 'center', px: '16px' }}>
        <Typography variant="caption" color="primary">
          $0.19
        </Typography>
        <Typography variant="caption" color="success.dark" ml="4px">
          (+0.01%)
        </Typography>
      </Box>
    </Box>
  );
}
