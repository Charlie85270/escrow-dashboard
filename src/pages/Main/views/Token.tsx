import { Box } from '@mui/material';
import React from 'react';
import tokenSvg from 'src/assets/token.svg';
import ViewTitle from 'src/components/ViewTitle';

export default function TokenView() {
  return (
    <Box mb="120px">
      <ViewTitle title="Token" iconUrl={tokenSvg} />
    </Box>
  );
}
