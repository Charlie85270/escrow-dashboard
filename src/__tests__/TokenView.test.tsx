import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { TokenView } from 'src/components/Token';

const mock = {
  address: '0xc748B2A084F8eFc47E086ccdDD9b7e67aEb571BF',
  scanner: 'test-scanner',
  transferEventCount: 2034,
  approvalEventCount: 1343,
  holders: 12345,
};

describe('when rendered CardTextBlock component', () => {
  it('should render passed prop `address`', () => {
    render(
      <TokenView
        address={mock.address}
        scanner={mock.scanner}
        transferEventCount={mock.transferEventCount}
        approvalEventCount={mock.approvalEventCount}
        holders={mock.holders}
      />
    );
    expect(screen.findByLabelText(mock.address)).toBeTruthy();
  });
  it('should render passed prop `scanner`', () => {
    render(
      <TokenView
        address={mock.address}
        scanner={mock.scanner}
        transferEventCount={mock.transferEventCount}
        approvalEventCount={mock.approvalEventCount}
        holders={mock.holders}
      />
    );
    expect(screen.findByLabelText(mock.scanner)).toBeTruthy();
  });
  it('should render passed prop `transferEventCount`', () => {
    render(
      <TokenView
        address={mock.address}
        scanner={mock.scanner}
        transferEventCount={mock.transferEventCount}
        approvalEventCount={mock.approvalEventCount}
        holders={mock.holders}
      />
    );
    expect(screen.getByText(mock.transferEventCount)).toBeInTheDocument();
  });

  it('should render passed prop `approvalEventCount`', () => {
    render(
      <TokenView
        address={mock.address}
        scanner={mock.scanner}
        transferEventCount={mock.transferEventCount}
        approvalEventCount={mock.approvalEventCount}
        holders={mock.holders}
      />
    );
    expect(screen.getByText(mock.approvalEventCount)).toBeInTheDocument();
  });
  it('should render passed prop `holders`', () => {
    render(
      <TokenView
        address={mock.address}
        scanner={mock.scanner}
        transferEventCount={mock.transferEventCount}
        approvalEventCount={mock.approvalEventCount}
        holders={mock.holders}
      />
    );
    expect(screen.getByText(mock.holders)).toBeInTheDocument();
  });
});

it('TokenView component renders correctly, corresponds to the snapshot', () => {
  const tree = renderer
    .create(
      <TokenView
        address={mock.address}
        scanner={mock.scanner}
        transferEventCount={mock.scanner}
        approvalEventCount={mock.approvalEventCount}
        holders={mock.holders}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
