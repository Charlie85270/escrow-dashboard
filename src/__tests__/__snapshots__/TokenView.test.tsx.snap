// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TokenView component renders correctly, corresponds to the snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1v5z4dq-MuiPaper-root-MuiCard-root"
>
  <div
    className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <div
      className="MuiBox-root css-xi606m"
    >
      <div
        className="MuiDivider-root MuiDivider-fullWidth MuiDivider-withChildren css-16qoh2c-MuiDivider-root"
        role="separator"
      >
        <span
          className="MuiDivider-wrapper css-qywfm8-MuiDivider-wrapper"
        >
          <p
            className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
          >
            Token Address
          </p>
        </span>
      </div>
      <a
        className="MuiTypography-root MuiTypography-inherit MuiTypography-alignCenter MuiLink-root MuiLink-underlineAlways css-1qm1xlu-MuiTypography-root-MuiLink-root"
        href="test-scanner/address/0xc748B2A084F8eFc47E086ccdDD9b7e67aEb571BF"
        onBlur={[Function]}
        onFocus={[Function]}
        rel="noreferrer"
        target="_blank"
      >
        0xc748B2A084F8eFc47E086ccdDD9b7e67aEb571BF
      </a>
    </div>
    <div
      className="MuiDivider-root MuiDivider-fullWidth MuiDivider-withChildren css-16qoh2c-MuiDivider-root"
      role="separator"
    >
      <span
        className="MuiDivider-wrapper css-qywfm8-MuiDivider-wrapper"
      >
        <p
          className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
        >
          Token Transfers
        </p>
      </span>
    </div>
    <p
      className="MuiTypography-root MuiTypography-body2 css-1ypfif-MuiTypography-root"
    >
      test-scanner
    </p>
    <div
      className="MuiDivider-root MuiDivider-fullWidth MuiDivider-withChildren css-16qoh2c-MuiDivider-root"
      role="separator"
    >
      <span
        className="MuiDivider-wrapper css-qywfm8-MuiDivider-wrapper"
      >
        <p
          className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
        >
          Token Approvals
        </p>
      </span>
    </div>
    <p
      className="MuiTypography-root MuiTypography-body2 css-1ypfif-MuiTypography-root"
    >
      1343
    </p>
    <div
      className="MuiDivider-root MuiDivider-fullWidth MuiDivider-withChildren css-16qoh2c-MuiDivider-root"
      role="separator"
    >
      <span
        className="MuiDivider-wrapper css-qywfm8-MuiDivider-wrapper"
      >
        <p
          className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
        >
          Token Holders
        </p>
      </span>
    </div>
    <p
      className="MuiTypography-root MuiTypography-body2 css-1ypfif-MuiTypography-root"
    >
      12345
    </p>
    <div
      className="MuiDivider-root MuiDivider-fullWidth MuiDivider-withChildren css-16qoh2c-MuiDivider-root"
      role="separator"
    >
      <span
        className="MuiDivider-wrapper css-qywfm8-MuiDivider-wrapper"
      >
        <p
          className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
        >
          Total Number Of Token Events
        </p>
      </span>
    </div>
    <p
      className="MuiTypography-root MuiTypography-body2 css-1ypfif-MuiTypography-root"
    >
      NaN
    </p>
  </div>
</div>
`;
